lunabot_embedded/firmware/teensy_main/teensy_main.ino

```diff
@@ -17,64 +19,66 @@ uint8_t buffer[64];
 uint8_t flags = 0;
 
 void recv() {
-  /* Create a stream that reads from the buffer. */
-  pb_istream_t stream = pb_istream_from_buffer(buffer, sizeof(buffer));
-
-  /* Now we are ready to decode the message. */
-  pb_decode(&stream, RobotEffort_fields, &effort);
-
-  actuation::cb(effort.lead_screw, effort.lin_act);
+  actuation::cb(effort.lin_act);
   drivetrain::cb(effort.left_drive, effort.right_drive);
   deposition::cb(effort.deposit);
   excavation::cb(effort.excavate);
 }
 
 void send() {
-  actuation::update(state.act_right_curr, state.lead_screw_curr, state.act_ang,
-                    state.lead_screw_ang);
+  actuation::update(state.act_right_curr);
   drivetrain::update(state.drive_left_curr, state.drive_right_curr, state.drive_left_ang,
                      state.drive_right_ang);
-  deposition::update(state.dep_curr, state.dep_ang);
+  deposition::update(state.dep_curr);
 
-  excavation::update(state.exc_curr);
+  excavation::update(state.exc_curr, state.exc_ang);
 
   uwb::update(state.uwb_dist_0, state.uwb_dist_1, state.uwb_dist_2);
 
   pb_ostream_t stream = pb_ostream_from_buffer(buffer, sizeof(buffer));
   pb_encode(&stream, RobotSensors_fields, &state);
 }
-IntervalTimer enc_timer;
+
 IntervalTimer uwb_timer;
 
 void setup() {
   Sabertooth_MotorCtrl::init_serial(ST_SERIAL, ST_BAUD_RATE);
   ACS711_Current_Bus::init_ads1115();
-  VLH35_Angle_Bus::init();
   M5Stack_UWB_Trncvr::init();
+  KillSwitchRelay::init();
 
-  enc_timer.begin(VLH35_Angle_Bus::transfer, ENC_TRANSFER_PERIOD);
   uwb_timer.begin(M5Stack_UWB_Trncvr::transfer, UWB_TRANSFER_PERIOD);
 
   // disable timeout
   MC1.setTimeout(0);
   MC2.setTimeout(0);
   MC3.setTimeout(0);
-  MC4.setTimeout(0);
 
   // set to fast ramp (1-10 - fast, 11-20 slow, 20-80 intermed)
   MC1.setRamping(1);
   MC2.setRamping(1);
   MC3.setRamping(1);
-  MC4.setRamping(1);
 }
 
 elapsedMillis ms_until_send;
+elapsedMillis ms_from_recv;
 elapsedMillis ms_curr_update;
 
 void loop() {
   int n;
   n = RawHID.recv(buffer, 0); // 0 timeout = do not wait
   if (n > 0) {
+    /* Create a stream that reads from the buffer. */
+    pb_istream_t stream = pb_istream_from_buffer(buffer, sizeof(buffer));
+
+    /* Now we are ready to decode the message. */
+    pb_decode(&stream, RobotEffort_fields, &effort);
+  }
+
+  if (ms_from_recv > RX_PERIOD) {
+    ms_until_recv -= RX_PERIOD;
+    // TODO, add timer if robot effort not changing for too long, exit?
+    KillSwitchRelay::logic(effort);
     recv();
```

Comment:
just for clarity, change this to CTRL instead of RX. similarly change the function name to ctrl().  recv happens in the `if(n>0)` block